server/src/main/java/io/druid/client/CachingClusteredClient.java

```diff
@@ -429,17 +431,12 @@ private void addSequencesFromServer(ArrayList<Sequence<T>> listOfSequences)
               final Sequence<T> resultSeqToAdd;
               if (!server.isAssignable() || !populateCache || isBySegment) { // Direct server queryable
                 if (!isBySegment) {
-                  resultSeqToAdd = clientQueryable.run(query.withQuerySegmentSpec(segmentSpec), responseContext);
+                  resultSeqToAdd = clientQueryable.run(queryPlus.withQuerySegmentSpec(segmentSpec), responseContext);
                 } else {
                   // bySegment queries need to be de-serialized, see DirectDruidClient.run()
-
-                  @SuppressWarnings("unchecked")
-                  final Query<Result<BySegmentResultValueClass<T>>> bySegmentQuery =
-                      (Query<Result<BySegmentResultValueClass<T>>>) ((Query) query);
-
                   @SuppressWarnings("unchecked")
                   final Sequence<Result<BySegmentResultValueClass<T>>> resultSequence = clientQueryable.run(
-                      bySegmentQuery.withQuerySegmentSpec(segmentSpec),
+                      queryPlus.withQuerySegmentSpec(segmentSpec),
```

Comment:
i think its same, bySegmentQuery was just a another name for query, but @leventov can confirm